src/Native/Unix/System.Native/pal_sysctl.cpp

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_config.h"
+
+// These functions are only used for platforms which support
+// using sysctl to gather system information.
+
+#if HAVE_SYS_SYSCTL_H
+
+#include "pal_utilities.h"
+#include "pal_sysctl.h"
+#include "pal_errno.h"
+#include "pal_safecrt.h"
+
+#include <errno.h>
+#include <memory>
+
+#include <sys/types.h>
+#include <sys/sysctl.h>
+
+#ifndef __linux__
+extern "C" int32_t SystemNative_SysctlByName(const char* name, void* value, size_t* len)
```

Comment:
Is this really needed? sysctlbyname() is fancy but slow, unless we have a good reason it should not be used.

Reply:
This is mostly for completness @janvorli. We already have code using private copies of this call. Since c# does not have easy access to .h enums and defines this gives you easy option to reference variables be name like sysctl("kern.osrelease")